lcls-twincat-optics/lcls_twincat_optics_plc/POUs/Main.TcPOU

```diff
@@ -0,0 +1,109 @@
+	fbMotionStage_m2 : FB_MotionStage;
+
+	M3 : DUT_MotionStage:=(nCommand:=3); // Xup
+	fbMotionStage_m3 : FB_MotionStage;
+
+	M4 : DUT_MotionStage:=(nCommand:=3); // Xdwn
+	fbMotionStage_m4 : FB_MotionStage;
+
+	M5 : DUT_MotionStage:=(nCommand:=3); // PitchCoarse
+	fbMotionStage_m5 : FB_MotionStage;
+
+	M6 : DUT_MotionStage:=(nCommand:=3); // Bender
+	fbMotionStage_m6 : FB_MotionStage;
+	
+	fbRunHOMS_M1L0 : FB_RunHOMS;
+	
+	bExecuteCoupleYM1L0 : BOOL;
+	bExecuteDecoupleYM1L0 : BOOL;
+	bExecuteCoupleXM1L0 : BOOL;
+	bExecuteDecoupleXM1L0 : BOOL;
+	
+	bGantryAlreadyCoupledYM1L0 : BOOL;
+	bGantryAlreadyCoupledXM1L0 : BOOL;
+	
+	nCurrGantryYM1L0 : LINT; // encoder counts = nm
+	nCurrGantryXM1L0 : LINT; // encoder counts = nm
+	nGantryTolYM1L0 : LINT := 50000; // encoder counts = nm
+	nGantryTolXM1L0 : LINT := 50000; // encoder counts = nm
+	
+	// M2L0
+	M7 : DUT_MotionStage:=(nCommand:=3); // Yup
+	fbMotionStage_m7 : FB_MotionStage;
+
+	M8 : DUT_MotionStage:=(nCommand:=3); // Ydwn
+	fbMotionStage_m8 : FB_MotionStage;
+
+	M9 : DUT_MotionStage:=(nCommand:=3); // Xup
+	fbMotionStage_m9 : FB_MotionStage;
+
+	M10 : DUT_MotionStage:=(nCommand:=3); // Xdwn
+	fbMotionStage_m10 : FB_MotionStage;
+
+	M11 : DUT_MotionStage:=(nCommand:=3); // PitchCoarse
+	fbMotionStage_m11 : FB_MotionStage;
+
+	M12 : DUT_MotionStage:=(nCommand:=3); // Bender
+	fbMotionStage_m12 : FB_MotionStage;
+
+	fbRunHOMS_M2L0 : FB_RunHOMS;
+
+	bExecuteCoupleYM2L0 : BOOL;
+	bExecuteDecoupleYM2L0 : BOOL;
+	bExecuteCoupleXM2L0 : BOOL;
+	bExecuteDecoupleXM2L0 : BOOL;
+	
+	bGantryAlreadyCoupledYM2L0 : BOOL;
+	bGantryAlreadyCoupledXM2L0 : BOOL;
+	
+	nCurrGantryYM2L0 : LINT; // encoder counts = nm
+	nCurrGantryXM2L0 : LINT; // encoder counts = nm
+	nGantryTolYM2L0 : LINT := 50000; // encoder counts = nm
+	nGantryTolXM2L0 : LINT := 50000; // encoder counts = nm
+END_VAR]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[// M1L0
+fbRunHOMS_M1L0(stYup:=M1, stYdwn:=M2, stXup:=M3, stXdwn:=M4, stPitch:=M5, stBender:=M6,
+	           nYupEncRef:=14760900, nYdwnEncRef:=13941900, nXupEncRef:=24435400, nXdwnEncRef:=25239600,
+			   bExecuteCoupleY:=bExecuteCoupleYM1L0, bExecuteCoupleX:=bExecuteCoupleXM1L0,
+			   bExecuteDecoupleY:=bExecuteDecoupleYM1L0, bExecuteDecoupleX:=bExecuteDecoupleXM1L0,
+			   nGantryTolY:=nGantryTolYM1L0, nGantryTolX:=nGantryTolXM1L0,
+               bGantryAlreadyCoupledY=>bGantryAlreadyCoupledYM1L0, bGantryAlreadyCoupledX=>bGantryAlreadyCoupledXM1L0,
+               nCurrGantryY=>nCurrGantryYM1L0, nCurrGantryX=>nCurrGantryXM1L0);
+
+fbMotionStage_m1(stMotionStage:=M1);
+fbMotionStage_m2(stMotionStage:=M2);
+fbMotionStage_m3(stMotionStage:=M3);
+fbMotionStage_m4(stMotionStage:=M4);
+fbMotionStage_m5(stMotionStage:=M5);
+fbMotionStage_m6(stMotionStage:=M6);
+
+// M2L0
+fbRunHOMS_M2L0(stYup:=M7, stYdwn:=M8, stXup:=M9, stXdwn:=M10, stPitch:=M11, stBender:=M12,
+	           nYupEncRef:=14897597, nYdwnEncRef:=14551600, nXupEncRef:=23972500, nXdwnEncRef:=24766500,
```

Comment:
Similarly, I think making constants out of the encoder reference values would be advisable. You can keep them all in one place and have better assurance they don't get accidentally changed if these lines get modified in the future.

Reply:
These global variables are more obvious and I added to `GVL_M1L0` and `GVL_M2L0`, which for now are in the library project but should actually live on specific optics PLC projects and get passed as input parameters to `FB_RunHOMS`.

Reply:
Great! I like that layout choice.

One minor thing - the `VAR_GLOBAL` section should be marked as `CONSTANT`, as I think you wouldn't want these to be changed online.

Reply:
Resolved in https://github.com/pcdshub/MR1L0-MR2L0/pull/1